### PR TITLE
Features/db fetch ssr

### PR DESCRIPTION
**Pull request by Ozanyurdakul98**

### CODE REVIEW

File: pages/search/index.js [hunk 1]
```diff
@@ -55,44 +45,57 @@ function Search(location) {
...
 }
-
 export default Search;
+
+export async function getServerSideProps(context) {
+	const query = context.query;
+	await db.connect();
+
+	const fetchingListings = await StudioListing.find();
+	const fetchedListings = JSON.parse(JSON.stringify(fetchingListings));
```
Comment (Ozanyurdakul98) [inline pages/search/index.js:93]: without this i get serialization error [Object, Object]. In the beginning i wanted here also a step to serialize the fetched data, by sorting out what fetched Object's should be propped. But every Objectvalue of that was undefined. So now i fetch everything from a listing and pass it to be filtered in Clientside.

**Comment by Ozanyurdakul98** [comment]
Userstory https://github.com/Ozanyurdakul98/capstone-project/issues/27

File: models/StudioListing.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import mongoose from 'mongoose';
+
+const studioListingSchema = new mongoose.Schema(
+	{
+		id: { type: String },
```
Comment (ThomasF85) [inline models/StudioListing.js:5]: Is there a reason for you to set an id here which is not required? Mongoose will automatically set an _id field for you.

File: pages/search/index.js (hunk 1 above)
Comment (ThomasF85) [reply pages/search/index.js:93]: If you only need a certain number of props, then only pass them to the client:
```
{prop1, prop2, ...} = fetchedListings;

return {
   props: {
      listings: {prop1, prop2, ...}
...

**Review by ThomasF85 — APPROVED** [review]
Looks good, but please check my comments. You have a probably unnecessary id field which you can remove from your model.

**Pull request merged**